File: app/src/main/java/com/example/jaguar/weather/presenter/WeatherHolder.java
package com.example.jaguar.weather.presenter;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import com.example.jaguar.weather.R;
import com.example.jaguar.weather.model.CallBackImage;
import com.example.jaguar.weather.model.WeatherObj;
import java.io.InputStream;

public class WeatherHolder extends RecyclerView.ViewHolder {
   private TextView weatherTextView;
   private ImageView weatherImageView;

    public WeatherHolder(final View itemView, final Context context) {
        super(itemView);
        weatherImageView = itemView.findViewById(R.id.imageWeather);
        weatherTextView = itemView.findViewById(R.id.hello);

        final Dialog dialog = new Dialog(context);
        dialog.setContentView(R.layout.activity_weather_full);
        final TextView text = dialog.findViewById(R.id.text);

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                text.setText(weatherTextView.getText());
                dialog.show();
            }
        });
    }
    public void bindCrime(WeatherObj wObj) {
        new ImageLoad(wObj.getImage(), new CallBackImage() {
            @Override
            public void Bind(Bitmap bitmap) {
                weatherImageView.setImageBitmap(bitmap);
            }
        }).execute();
        weatherTextView.setText(wObj.getText());
    }
    class ImageLoad extends AsyncTask<String, Void, Bitmap> {
        private final String image;
        private final CallBackImage call;

        ImageLoad(String image, CallBackImage call) {
            this.image = image;
            this.call = call;
        }

        protected Bitmap doInBackground(String... strings) {
            Bitmap mIcon11 = null;
            try {
                InputStream in = new java.net.URL(image).openStream();
                mIcon11 = BitmapFactory.decodeStream(in);
            } catch (Exception e) {
                e.printStackTrace();
            }
            return mIcon11;
        }

        protected void onPostExecute(Bitmap result) {
            call.Bind(result);
        }
    }
}

File: app/src/main/java/com/example/jaguar/weather/adapters/WeatherHolder.kt
package com.example.jaguar.weather.adapters

import android.app.Dialog
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.jaguar.weather.R
import com.example.jaguar.weather.common.WeatherObj
import kotlinx.android.synthetic.main.activity_weather_full.*
import kotlinx.android.synthetic.main.list_item_weather.view.*

class WeatherHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
    private var weatherTextView: TextView
    private var weatherCondition: TextView
    private var imageWeather: ImageView
    private var dateR: TextView

    init {
        weatherTextView = this.itemView.hello
        imageWeather = this.itemView.imgV
        weatherCondition = this.itemView.condition
        dateR = this.itemView.date

        var dialog = Dialog(this.itemView.context)
        dialog.setContentView(R.layout.activity_weather_full)

        val text:TextView = dialog.text
        itemView?.setOnClickListener {
            text.setText(weatherTextView.text)
            dialog.show()
        }
    }

    fun bindCrime(wObj: WeatherObj) {
        weatherTextView.text = wObj.text
        Glide.with(itemView.getContext()).load(wObj.imgUrl1).override(100).into(imageWeather)
        weatherCondition.text = wObj.weatherCondition
        dateR.text = wObj.date
    }

}